blocks/rich-text/index.js

```diff
@@ -478,8 +479,8 @@ export class RichText extends Component {
 		const rootNode = this.editor.getBody();
 
 		if (
-			( event.keyCode === BACKSPACE && isHorizontalEdge( rootNode, true ) ) ||
-			( event.keyCode === DELETE && isHorizontalEdge( rootNode, false ) )
+			( event.keyCode === BACKSPACE && isVerticalEdge( rootNode, true ) && isHorizontalEdge( rootNode, true ) ) ||
```

Comment:
> Also, another issue I found related to #5159 is that we want to trigger onSplit only if we click delete at the end of RichText or backspace at the beginning. So we need to check both the vertical and the horizontal edge in both cases.

Huh, horizontal edge is only supposed to check  the first and last caret position? Not every horizontal line edge? So I don't understand why this is needed?

Reply:
Oh I see, this was not clear for me. updated